### ISSUE TITLE
Hotend Fans stop when GLCD reset button is pressed

### ISSUE THREAD

**Issue by bruce356** [KevinOConnor/klipper#805]
Hi, when using this method of energy stop/reset the fans also stop.
It can be a problem with hot ends cooled by fan like E3D v6 due to heat creep. 
The fans do kick back in after Klipper/Octoprint is re-booted, but sometimes operator forgetfulness plays a part.

Is there any way that the fans could simply be kept on or running. If Klipper/Octoprint were to automatically re-boot after a hard reset that would be ideal, instead of being left in a hung state.

Using a Cartesian printer with RUMBA board and RPi3
[klippy.zip](https://github.com/KevinOConnor/klipper/files/2499905/klippy.zip)

Thanks all contributors for the great firmware.
Regards - Bruce

**Comment by FHeilmann** [comment]
Fans have a so-called shutdown value for exactly this purpose I believe.
See these examples:
`[heater_fan]`:https://github.com/KevinOConnor/klipper/blob/master/config/example-extras.cfg#L365
`[temperature_fan]`https://github.com/KevinOConnor/klipper/blob/master/config/example-extras.cfg#L393
`[fan]`:https://github.com/KevinOConnor/klipper/blob/master/config/example.cfg#L237

**Comment by bruce356** [comment]
@FHeilmann  thanks for the reply, none of the recommended above links solve this problem. It is the Pi / Octoprint that needs to be reset, restarted, for hot end fans to restart (not worried about the Print cooling fan, that does not matter). Marlin will restart the fans after a hard reset without going into a bootloop, I know Klippers functionality is different but this should be possible for only the fans to restart, not the rest of the printing processes as these are halted anyway.

Sorry for the multi posts, I posted this one first and for some reason it did not show up as it normally should so I thought it had been lost somewhere and I posted again with different wording.
Regards - bruce

**Comment by FHeilmann** [comment]
I had a look at your klippy. log and there is nothing suspicious in there. At the very least the log you provide didn't capture any of the issues you describe. If you press the reset button on the LCD the fans will shut off, but so will the heater. The heatcreep you worry about will thus be a non-issue. I still don't understand the need to reboot the pi when you press the reset button on the LCD. All that button is supposed to do is reset the rumba board, not the pi. Issuing a FIRMWARE_RESTART from octopi should allow the rumba to re-configure itself and continue operation.

If this is not the case for you, there is probably some other issue that is out of the responsibility of klipper.

Finally, if this issue is a duplicate for the other one you can close the other one to reduce the clutter in the issue section.

**Comment by LazaroFilm** [comment]
You could also wire your cooling fan directly to a separate power line bypassing your controller board and have it power on/off when your turn your printer on/off.

**Comment by bruce356** [comment]
@FHeilmann hi, I am only trying to suggest an improvement in functionality for Klipper, yes as you suggest  the hard reset does stop all functionality (as it should) a "restart" or "firmware_restart" command does return operation to normal (obviously printing is halted and fans start up again if temp of hot end is above the preset 50 deg.). It would be far more convenient if this occurred automatically, but if it is not possible.
Thanks and regards - bruce

**Comment by jakep82** [comment]
Even in the case of an emergency stop (M112), the hotend fan should continue running as long as the MCU has power.   It would be bad to have the MCU automatically restart in the case of an emergency stop or other failure as a hardware malfunction might need user intervention to avoid a dangerous situation.  If the hotend fan is not running after an emergency stop, then you're either cutting power to the MCU in which case the fans can't run because the mosfets won't receive a signal, or you are experiencing a bug in the software.

**Comment by KevinOConnor** [comment]
Alas, some boards wire the "kill button" on the lcd headers directly to the mcu reset line.  (For example, the MKS Gen-L board that I have does this.)  This is an unfortunate choice and there's little the software can do to improve the error handling.  If the button is pressed, the software running on the micro-controller is halted without any notice and the software running on the host machine also does not get any notice.  Eventually the host software will detect that the micro-controller is no longer responding, but it has no knowledge on why the micro-controller is not functioning.  Attempting to reprogram the micro-controller in that situation could lead to worse results (eg, if the reason for the micro-controller failure is due to some systemic problem).

In short, the best advice I have is to avoid using the kill button on these boards.

-Kevin

**Comment by jakep82** [comment]
Another less elegant solution would be to separate out this wire from the ribbon cable, and connect it to a different pin on the board.
